fvp-psa-sp.mk

```diff
@@ -0,0 +1,41 @@
+FVP_USE_BASE_PLAT		?= y
+FVP_VIRTFS_ENABLE		?= y
+FVP_VIRTFS_AUTOMOUNT		?= y
+MEASURED_BOOT			?= y
+MEASURED_BOOT_FTPM		?= n
+TS_UEFI_TESTS			?= y
+
+TF_A_FLAGS ?= \
+	BL32=$(OPTEE_OS_PAGER_V2_BIN) \
+	BL33=$(EDK2_BIN) \
+	PLAT=fvp \
+	SPD=spmd \
+	SPMD_SPM_AT_SEL2=0 \
+	ARM_SPMC_MANIFEST_DTS=$(ROOT)/build/fvp/spmc_manifest.dts
+
+include fvp.mk
+include trusted-services.mk
+
+OPTEE_OS_COMMON_EXTRA_FLAGS += \
+	CFG_SECURE_PARTITION=y \
+	CFG_CORE_SEL1_SPMC=y \
+	CFG_CORE_HEAP_SIZE=131072 \
+	CFG_MAP_EXT_DT_SECURE=y
+
+# SP_COMMON_FLAGS is passed to each SP
+SP_COMMON_FLAGS ?=
+SP_PSA_ITS_EXTRA_FLAGS ?=
+SP_PSA_PS_EXTRA_FLAGS ?=
+SP_PSA_CRYPTO_EXTRA_FLAGS ?=
+SP_PSA_ATTESTATION_EXTRA_FLAGS ?=
+SP_SMM_GATEWAY_EXTRA_FLAGS ?=
+
+$(eval $(call build-sp,internal-trusted-storage,dc1eef48-b17a-4ccf-ac8b-dfcff7711b14,$(SP_PSA_ITS_EXTRA_FLAGS)))
+$(eval $(call build-sp,protected-storage,751bf801-3dde-4768-a514-0f10aeed1790,$(SP_PSA_PS_EXTRA_FLAGS)))
+$(eval $(call build-sp,crypto,d9df52d5-16a2-4bb2-9aa4-d26d3b84e8c0,$(SP_PSA_CRYPTO_EXTRA_FLAGS)))
+ifeq ($(MEASURED_BOOT),y)
```

Comment:
In theory there are no conflicts between the two, but I didn't test it in practice.

Reply:
Shall we add an option like MEASURED_BOOT_FTPM for IAT? Now measured boot always enables IAT while FTPM is optional. Should we implement something like:
MB FTPM IAT
0     0        0
0     x        x  not possible as FTPM and IAT should force MB to 1. 
1     0        0  shall this be invalid?
1     0       1
1     1       0
1     1       1 error not tested yet?

Reply:
> Now measured boot always enables IAT

This is only true if the `fvp-psa-sp.mk` top level Makefile is used. In this case for me it doesn't seem reasonable to try disabling IAT and enabling fTPM.

The way I see it, if

- `fvp-psa-sp.mk` is used: `MEASURED_BOOT=y` enables the event log handling/forwarding plus the default consumer of the event log, which is IAT. Optionally you can enable fTPM too.
- `fvp.mk` is used:  `MEASURED_BOOT=y` enables the event log handling/forwarding plus the default consumer of the event log, which is fTPM. You cannot enable IAT since in this case you don't have FF-A, SPMC, etc.

Reply:
Ahh, I see. Thanks for explaining. It's all right then.